include/common/mpmc_queue.h

```diff
@@ -30,28 +30,9 @@ constexpr int DEFAULT_SIZE = 100;
 
 template <typename T>
 class mpmc_queue{
-   private:
-      int size;
-      std::vector<T> buffer;
-      std::atomic<unsigned long long> pread;
-      std::atomic<unsigned long long> pwrite;
-      std::atomic<unsigned long long> internal_pwrite;
-      std::atomic<unsigned long long> internal_pread;
-
-      bool lockfree = false;
-
-      std::mutex mutex;
-      std::condition_variable emptycv;
-      std::condition_variable fullcv;
-
-      bool full(unsigned long long current);
-      bool empty(unsigned long long current);
-   public:  
+   public:
       typedef T value_type;
-      T pop();
-      bool push(T item);
-      bool empty();
-      
+
```

Comment:
This typedef is used to determine the type of the queue contents in some patterns.